doc/source/main_install.rst

```diff
@@ -1,221 +1,182 @@
 
 
-.. _install-buildstream:
-
-Installing BuildStream
-----------------------
-
-Installing from PyPI
-~~~~~~~~~~~~~~~~~~~~
-Once you have the base system dependencies, you can install the BuildStream
-python package as a regular user.
-
-To install from PyPI, you will additionally require:
-
- - pip for python3 (only required for setup)
- - Python 3 development libraries and headers
-
-
-For the latest dev snapshot of BuildStream 2, simply run the following command::
-
-    pip3 install --user --pre BuildStream
-
-This will install latest dev snapshot of BuildStream and its pure python
-dependencies into your user's homedir in ``~/.local``.
-
-.. note::
 
-   At time of writing, BuildStream 2 is only available as dev snapshots; this
-   is why the ``--pre`` option is required.  Running
-   ``pip3 install --user BuildStream`` (without the ``--pre`` option)
-   will install Buildsteam 1.
-
-You can also install a specific dev snapshot of Buildstream by specifying the
-version. eg ``pip3 install --user BuildStream==1.93.2.dev0``.
-Available versions can be found on the BuildStream history page `on PyPi 
-<https://pypi.org/project/BuildStream/#history>`_.
-Note that some of the oldest versions are not available on PyPI.
-
-Keep following the :ref:`instructions below<post-install>` to ensure that the ``bst``
-command is in your ``PATH``.
-
-Upgrading from PyPI
-+++++++++++++++++++
-Once you have already installed BuildStream from PyPI, you can later update
-to the latest dev snapshot like so::
-
-
-    pip3 install --user --upgrade --pre BuildStream
-
-
-
-Installing from a git checkout
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-To install directly from the `git repository <https://github.com/apache/buildstream>`_
-using python's ``pip`` package manager, you will additionally require:
-
-- pip for python3 (only required for setup)
-- Python 3 development libraries and headers
-- git (to checkout BuildStream)
+.. _install-buildstream:
 
-Before installing, please check the existing tags in the git repository
-and determine which version you want to install.
+Installing BuildStream from a git checkout
+------------------------------------------
 
-Run the following commands::
+First, clone the repository. Please check the existing tags in the git
+repository and determine which version you want to install::
 
 
     git clone https://github.com/apache/buildstream.git
     cd buildstream
     git checkout <desired release tag>
+
+We recommend ``pip`` as a frontend to the underlying ``setuptools`` build
+system.  The following command will build and install BuildStream into your
+user's homedir in ``~/.local``, and will attempt to fetch and install any
+required PyPI dependencies from the internet at the same time::
+
+
     pip3 install --user .
 
-This will install BuildStream's pure python dependencies into
-your user's homedir in ``~/.local`` and will run BuildStream directly
-from the git checkout directory.
+You can optionally use Pip's
```

Comment:
I don't believe the `-e` option has worked for a very long time (with buildstream specifically, this might have been since the transition to use cython).

Reply:
OK, will remove

Reply:
Opened https://github.com/apache/buildstream/issues/1760 so we have somewhere to answer the question "Why can't I use `pip -e`" and will link that from the docs